aptos-move/move-examples/drand/sources/drand.move

```diff
@@ -0,0 +1,136 @@
+/// corresponding to `pk`.
+///
+/// Rounds happen once every 3 seconds (for the "bls-unchained-on-g1" beacon). This way, given a UNIX timestamp, one can
+/// easily derive the round # `i` that drand should have signed to produce randomness for that round.
+///
+/// The parameters of the "bls-unchained-on-g1" drand beacon, which are hardcoded in this module, were obtained from
+/// querying [the drand REST API](https://api.drand.sh/dbd506d6ef76e5f386f41c651dcb808c5bcbd75471cc4eafa3f4df7ad4e4c493/info).
+///
+/// ```
+/// {
+///     "public_key": "a0b862a7527fee3a731bcb59280ab6abd62d5c0b6ea03dc4ddf6612fdfc9d01f01c31542541771903475eb1ec6615f8d0df0b8b6dce385811d6dcf8cbefb8759e5e616a3dfd054c928940766d9a5b9db91e3b697e5d70a975181e007f87fca5e",
+///     "period": 3,
+///     "genesis_time": 1677685200,
+///     "hash": "dbd506d6ef76e5f386f41c651dcb808c5bcbd75471cc4eafa3f4df7ad4e4c493",
+///     "groupHash": "a81e9d63f614ccdb144b8ff79fbd4d5a2d22055c0bfe4ee9a8092003dab1c6c0",
+///     "schemeID": "bls-unchained-on-g1",
+///     "metadata": {"beaconID": "fastnet"}
+/// }
+/// ```
+
+module drand::drand {
+    use std::hash::{sha3_256, sha2_256};
+    use std::option::{Self, Option, extract};
+    use std::vector;
+    use std::error;
+    use aptos_std::crypto_algebra::{eq, pairing, one, deserialize, hash_to, from_u64, serialize};
+    use aptos_std::bls12381_algebra::{G1, G2, Gt, FormatG2Compr, FormatG1Compr, HashG1XmdSha256SswuRo, Fr, FormatFrMsb};
+
+    /// The `bls-unchained-on-g1` drand beacon produces an output every 3 seconds.
+    /// (Or goes into catchup mode, if nodes fall behind.)
+    const PERIOD_SECS : u64 = 3;
+
+    /// The UNIX time (in seconds) at which the beacon started operating (this is the time of round #1)
+    const GENESIS_TIMESTAMP : u64 = 1677685200;
+
+    /// The drand beacon's PK, against which any beacon output for a round `i` can be verified.
+    const DRAND_PUBKEY : vector<u8> = x"a0b862a7527fee3a731bcb59280ab6abd62d5c0b6ea03dc4ddf6612fdfc9d01f01c31542541771903475eb1ec6615f8d0df0b8b6dce385811d6dcf8cbefb8759e5e616a3dfd054c928940766d9a5b9db91e3b697e5d70a975181e007f87fca5e";
+
+    /// The domain-separation tag (DST) used in drand's BLS signatures
+    const DRAND_DST: vector<u8> = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
+
+    /// Error code for when anyone submits an incorrect randomness in our APIs (e.g., wrong-size).
+    const E_INCORRECT_RANDOMNESS: u64 = 1;
+
+    /// A randomness object, created from a successfully-verified piece of drand randomness.
+    /// This object can be converted into a uniform random integer.
+    struct Randomness has drop {
+        bytes: vector<u8>
+    }
+
+    /// Checks if the randomness in `bytes` verifies for the specified `round`.
+    /// If it verifies, returns the actual randomness, which is a hash function applied over `bytes`.
+    public fun verify_and_extract_randomness(
+        bytes: vector<u8>,
```

Comment:
why not call this signature? it's confusing with the fact you are returning "bytes" as  a field of randomness but it's certainly another "bytes"

Reply:
Fixed.